k8s-scheduler/src/test/java/org/dcm/WorkloadGeneratorIT.java

```diff
@@ -208,8 +226,17 @@ void runTrace(final DefaultKubernetesClient client, final String fileName, final
                 final float mem = Float.parseFloat(parts[5].replace(">", "")) / memScaleDown;
                 final int vmCount = Integer.parseInt(parts[6].replace(">", ""));
 
+                boolean createAffinityRequirements = false;
+                if (r.nextInt(numberOfDeployments)
```

Comment:
Make sure to initialize Random() with a deterministic seed, so that the same deployments across runs are sampled for being tagged with the affinity requirements.